pkg/operator/staticpod/controllers.go

```diff
@@ -0,0 +1,63 @@
+package staticpod
+
+import (
+	"k8s.io/client-go/informers"
+	"k8s.io/client-go/kubernetes"
+
+	"github.com/openshift/library-go/pkg/operator/staticpod/controller/common"
+	"github.com/openshift/library-go/pkg/operator/staticpod/controller/deployment"
+	"github.com/openshift/library-go/pkg/operator/staticpod/controller/installer"
+	"github.com/openshift/library-go/pkg/operator/staticpod/controller/node"
+)
+
+type staticPodOperatorControllers struct {
+	deploymentController *deployment.DeploymentController
+	installerController  *installer.InstallerController
+	nodeController       *node.NodeController
+}
+
+// NewControllers provides all control loops needed to run a static pod based operator. That includes:
+// 1. DeploymentController - this watches multiple resources for "latest" input that has changed from the most current deploymentID.
+//    When a change is found, it creates a new deployment by copying resources and adding the deploymentID suffix to the names
+//    to make a theoretically immutable set of deployment data.  It then bumps the latestDeploymentID and starts watching again.
+// 2. InstallerController - this watches the latestDeploymentID and the list of kubeletStatus (alpha-sorted list).  When a latestDeploymentID
+//    appears that doesn't match the current latest for first kubeletStatus and the first kubeletStatus isn't already transitioning,
+//    it kicks off an installer pod.  If the next kubeletStatus doesn't match the immediate prior one, it kicks off that transition.
+// 3. NodeController - watches nodes for master nodes and keeps the operator status up to date
+func NewControllers(targetNamespaceName string, command, deploymentConfigMaps, deploymentSecrets []string,
+	staticPodOperatorClient common.OperatorClient, kubeClient kubernetes.Interface, kubeInformersNamespaceScoped,
+	kubeInformersClusterScoped informers.SharedInformerFactory) *staticPodOperatorControllers {
+	controller := &staticPodOperatorControllers{}
+
+	controller.deploymentController = deployment.NewDeploymentController(
+		targetNamespaceName,
+		deploymentConfigMaps,
+		deploymentSecrets,
+		kubeInformersNamespaceScoped,
+		staticPodOperatorClient,
+		kubeClient,
+	)
+
+	controller.installerController = installer.NewInstallerController(
+		targetNamespaceName,
+		deploymentConfigMaps,
+		deploymentSecrets,
+		command,
+		kubeInformersNamespaceScoped,
+		staticPodOperatorClient,
+		kubeClient,
+	)
+
+	controller.nodeController = node.NewNodeController(
+		staticPodOperatorClient,
+		kubeInformersClusterScoped,
+	)
+
+	return controller
+}
+
+func (o *staticPodOperatorControllers) Run(stopCh <-chan struct{}) {
```

Comment:
This call should block.  And please give glogs

Reply:
glog info is given by all individual controller, do you need more glog?

sure I can make it block